module_auto_update/models/module.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import logging
+
+from odoo import api, fields, models
+from odoo.modules.module import get_module_path
+
+_logger = logging.getLogger(__name__)
+try:
+    from checksumdir import dirhash
+except ImportError:
+    _logger.debug('Cannot `import checksumdir`.')
+
+
+class Module(models.Model):
+    _inherit = 'ir.module.module'
+
+    checksum_installed = fields.Char(
+        string="Installed Module Checksum",
+    )
+    checksum_dir = fields.Char(
+        string="Current Module Checksum",
+        compute='_compute_checksum_dir',
+    )
+    upgrade_available = fields.Boolean(
+        default=False,
+    )
+
+    @api.multi
+    def _compute_checksum_dir(self):
+        for r in self:
+            r.checksum_dir = dirhash(get_module_path(r.name), 'sha1')
+
+    @api.multi
+    def button_immediate_upgrade(self):
+        res = super(Module, self).button_immediate_upgrade()
+        self.post_upgrade_actions()
+        return res
+
+    @api.model
+    def check_upgrade_availability(self):
+        self.update_list()
+        self.determine_upgradeability()
+
+    @api.depends('checksum_dir')
```

Comment:
It does not; you can only edit computed fields with depends methods. They technically run, but the changes aren't stored to DB because you are in the cache layer.